pkg/scheduler/scheduler.go

```diff
@@ -150,6 +155,102 @@ type Config struct {
 	DisablePreemption bool
 }
 
+type schedulerOptions struct {
+	schedulerName                  string
+	hardPodAffinitySymmetricWeight int32
+	enableEquivalenceClassCache    bool
+	disablePreemption              bool
+	percentageOfNodesToScore       int32
+	bindTimeoutSeconds             int64
+	schedulerAlgorithmSource       kubeschedulerconfig.SchedulerAlgorithmSource
+}
+
+// Option is function for schedulerOptions to set value
+type Option func(*schedulerOptions)
+
+// WithSchedulerName set schedulerName for Scheduler
+func WithSchedulerName(r string) Option {
+	return func(o *schedulerOptions) {
+		o.schedulerName = r
+	}
+}
+
+// WithHardPodAffinitySymmetricWeight set hardPodAffinitySymmetricWeight for Scheduler
+func WithHardPodAffinitySymmetricWeight(r int32) Option {
+	return func(o *schedulerOptions) {
+		o.hardPodAffinitySymmetricWeight = r
+	}
+}
+
+// WithEnableEquivalenceClassCache set enableEquivalenceClassCache for Scheduler
+func WithEnableEquivalenceClassCache(r bool) Option {
```

Comment:
To make this easier to read, could you change this to `WithEquivalenceClassCacheEnabled`? Similarly for DisablePreemption.